src/main/java/duke/DeleteCommand.java

```diff
@@ -0,0 +1,37 @@
+package duke;
+
+import java.io.IOException;
+
+/**
+ * Represents a delete command.
+ */
+public class DeleteCommand extends Command {
+
+    /** User input as a String */
+    String userInput;
+
+    /**
+     * Constructs a DeleteCommand object with user input.
+     *
+     * @param userInput User input as a String.
+     */
+    DeleteCommand(String userInput) {
+        this.userInput = userInput;
+    }
+
+    /**
+     * Executes the deleting of a task.
+     *
+     * @param taskList Task list containing tasks.
+     * @param storage Storage for storing and retrieving all tasks.
+     * @param ui Handles printing of user interaction.
+     * @throws DukeException When input for delete is invalid, respective error messages are printed.
+     * @throws IOException When input for delete is invalid, respective error messages are printed.
+     */
+    @Override
+    public void execute(
```

Comment:
Consider changing the method name to be more explanatory, so to avoid ambiguity between other execute methods from other classes.

Reply:
It is an abstract method so it has to be named 'execute' for all Command classes. Sorry for the confusion!